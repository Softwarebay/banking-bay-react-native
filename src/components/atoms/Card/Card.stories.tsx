import { action } from '@storybook/addon-actions';
import { color, object, text } from '@storybook/addon-knobs';
import { storiesOf } from '@storybook/react-native';

import { Heading, IconButton } from 'components/atoms';
import { ViewStory } from 'utils/storybook/ViewStory';

import { colors } from 'theme';
import { Card } from './Card';

const cardStyleKnob = {
	height: 60,
	paddingHorizontal: 20,
	paddingVertical: 14,
};

storiesOf('Atoms/Card', module)
	.addDecorator((getStory) => <ViewStory>{getStory()}</ViewStory>)
	.add('With text and icon button', () => (
		<Card
			style={object('style', cardStyleKnob)}
			testID={text('testID', 'Card-test-id')}
		>
			<Heading
				size="h6"
				style={{ flex: 1 }}
				testID="heading-test-id"
				text="Text inside Card"
			/>
			<IconButton
				iconColor={color('iconColor', colors.mainDark)}
				iconName="check"
				testID="IconButton-test-id"
				onPress={action('onPressIconButton')}
			/>
		</Card>
	));
