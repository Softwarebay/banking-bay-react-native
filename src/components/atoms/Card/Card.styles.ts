import { Platform, StyleSheet } from 'react-native';

import { colors, shadows } from 'theme';

export const styles = StyleSheet.create({
	card: {
		backgroundColor: colors.white,
		borderRadius: 10,
		...Platform.select({
			android: {
				...shadows.android,
			},
			ios: {
				...shadows.ios,
			},
		}),
	},
});
