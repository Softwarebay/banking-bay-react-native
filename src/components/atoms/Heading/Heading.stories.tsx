import { color, text } from '@storybook/addon-knobs';
import { storiesOf } from '@storybook/react-native';
import { colors } from 'theme';
import { ViewStory } from 'utils/storybook/ViewStory';

import { Heading } from './Heading';

storiesOf('Atoms/Heading', module)
	.addDecorator((getStory) => <ViewStory center>{getStory()}</ViewStory>)
	.add('h1', () => (
		<Heading
			color={color('color', colors.mainDark)}
			size="h1"
			testID={text('testID', 'Heading-test-id')}
			text={text('title', 'Heading')}
		/>
	))
	.add('h2', () => (
		<Heading
			color={color('color', colors.mainDark)}
			size="h2"
			testID={text('testID', 'Heading-test-id')}
			text={text('title', 'Heading')}
		/>
	))
	.add('h3', () => (
		<Heading
			color={color('color', colors.mainDark)}
			size="h3"
			testID={text('testID', 'Heading-test-id')}
			text={text('title', 'Heading')}
		/>
	))
	.add('h4', () => (
		<Heading
			color={color('color', colors.mainDark)}
			size="h4"
			testID={text('testID', 'Heading-test-id')}
			text={text('title', 'Heading')}
		/>
	))
	.add('h5', () => (
		<Heading
			color={color('color', colors.mainDark)}
			size="h5"
			testID={text('testID', 'Heading-test-id')}
			text={text('title', 'Heading')}
		/>
	))
	.add('h6', () => (
		<Heading
			color={color('color', colors.mainDark)}
			size="h6"
			testID={text('testID', 'Heading-test-id')}
			text={text('title', 'Heading')}
		/>
	))
	.add('h3 success', () => (
		<Heading
			color={color('color', colors.green)}
			size="h3"
			testID={text('testID', 'Heading-test-id')}
			text={text('title', 'Heading')}
		/>
	))
	.add('h3 error', () => (
		<Heading
			color={color('color', colors.pink)}
			size="h3"
			testID={text('testID', 'Heading-test-id')}
			text={text('title', 'Heading')}
		/>
	))
	.add('h3 long text', () => (
		<Heading
			color={color('color', colors.mainDark)}
			size="h3"
			testID={text('testID', 'Heading-test-id')}
			text={text(
				'title',
				'Lorem ipsum dolor sit amet, consectetur adipiscing elit'
			)}
		/>
	));
