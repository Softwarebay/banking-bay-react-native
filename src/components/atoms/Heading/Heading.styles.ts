import { StyleSheet } from 'react-native';

import { colors } from 'theme';

export const styles = StyleSheet.create({
	default: {
		color: colors.mainDark,
	},
	h1: {
		fontSize: 44,
		fontWeight: '700',
		lineHeight: 48.4,
	},
	h2: {
		fontSize: 36,
		fontWeight: '700',
		lineHeight: 39.6,
	},
	h3: {
		fontSize: 28,
		fontWeight: '700',
		lineHeight: 30.8,
	},
	h4: {
		fontSize: 20,
		fontWeight: '500',
		lineHeight: 24,
	},
	h5: {
		fontSize: 16,
		fontWeight: '600',
		lineHeight: 20.8,
	},
	h6: {
		fontSize: 14,
		fontWeight: '600',
		lineHeight: 22.4,
	},
});
