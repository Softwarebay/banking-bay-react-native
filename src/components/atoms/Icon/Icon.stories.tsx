import { color, number } from '@storybook/addon-knobs';
import { storiesOf } from '@storybook/react-native';
import { colors } from 'theme';
import { ViewStory } from 'utils/storybook/ViewStory';

import { Icon } from './Icon';

storiesOf('Atoms/Icon', module)
	.addDecorator((getStory) => (
		<ViewStory style={{ backgroundColor: colors.screenBackground }} center>
			{getStory()}
		</ViewStory>
	))
	.add('Logo', () => (
		<Icon
			color={color('color', colors.mainDark)}
			name="logo"
			size={number('size', 20)}
		/>
	))
	.add('Facebook', () => (
		<Icon
			color={color('color', colors.mainDark)}
			name="facebook"
			size={number('size', 20)}
		/>
	))
	.add('Twitter', () => (
		<Icon
			color={color('color', colors.mainDark)}
			name="twitter"
			size={number('size', 20)}
		/>
	))
	.add('Google+', () => (
		<Icon
			color={color('color', colors.mainDark)}
			name="google-plus"
			size={number('size', 20)}
		/>
	))
	.add('Check', () => (
		<Icon
			color={color('color', colors.mainDark)}
			name="check"
			size={number('size', 20)}
		/>
	))
	.add('Eye off', () => (
		<Icon
			color={color('color', colors.mainDark)}
			name="eye-off"
			size={number('size', 20)}
		/>
	));
