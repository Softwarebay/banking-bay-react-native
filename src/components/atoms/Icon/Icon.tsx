import {
	Amazon,
	Atm,
	Check,
	ChevronLeft,
	CreditCard,
	CreditCards,
	Deposit,
	DollarSign,
	Ebay,
	EyeOff,
	Facebook,
	FlagUSA,
	GooglePlus,
	Home,
	Info,
	LoanWallet,
	Logo,
	MoneyTransfer,
	MoreTiles,
	NotificationBell,
	Paypal,
	ProfileIcon,
	Repeat,
	Report,
	Search,
	Smartphone,
	Twitter,
} from 'assets/icons';

export interface Props {
	name:
		| 'facebook'
		| 'twitter'
		| 'check'
		| 'eye-off'
		| 'google-plus'
		| 'logo'
		| 'chevron-left'
		| 'credit-card'
		| 'dollar-sign'
		| 'repeat'
		| 'smartphone'
		| 'profile'
		| 'credit-cards'
		| 'deposits'
		| 'loans'
		| 'notifications'
		| 'dashboard'
		| 'more'
		| 'amazon'
		| 'money-transfer'
		| 'paypal'
		| 'flag-usa'
		| 'search'
		| 'atm'
		| 'ebay'
		| 'home'
		| 'info';
	size: number;
	color?: string;
}

// eslint-disable-next-line consistent-return
export const Icon = ({ name, color, size }: Props) => {
	switch (name) {
		case 'facebook':
			return (
				<Facebook
					color={color}
					height={size}
					testID="facebook-icon-test-id"
					width={size}
				/>
			);
		case 'google-plus':
			return (
				<GooglePlus
					color={color}
					height={size}
					testID="google-plus-icon-test-id"
					width={size}
				/>
			);
		case 'twitter':
			return (
				<Twitter
					color={color}
					height={size}
					testID="twitter-icon-test-id"
					width={size}
				/>
			);
		case 'logo':
			return (
				<Logo
					color={color}
					height={size}
					testID="logo-icon-test-id"
					width={size}
				/>
			);
		case 'check':
			return (
				<Check
					color={color}
					height={size}
					testID="check-icon-test-id"
					width={size}
				/>
			);
		case 'eye-off':
			return (
				<EyeOff
					color={color}
					height={size}
					testID="eye-off-icon-test-id"
					width={size}
				/>
			);
		case 'chevron-left':
			return (
				<ChevronLeft
					color={color}
					height={size}
					testID="chevron-left-icon-test-id"
					width={size}
				/>
			);
		case 'credit-card':
			return (
				<CreditCard
					color={color}
					height={size}
					testID="credit-card-icon-test-id"
					width={size}
				/>
			);
		case 'smartphone':
			return (
				<Smartphone
					color={color}
					height={size}
					testID="smartphone-icon-test-id"
					width={size}
				/>
			);
		case 'dollar-sign':
			return (
				<DollarSign
					color={color}
					height={size}
					testID="dollar-sign-icon-test-id"
					width={size}
				/>
			);
		case 'repeat':
			return (
				<Repeat
					color={color}
					height={size}
					testID="repeat-icon-test-id"
					width={size}
				/>
			);
		case 'profile':
			return (
				<ProfileIcon
					color={color}
					height={size}
					testID="profile-icon-test-id"
					width={size}
				/>
			);
		case 'credit-cards':
			return (
				<CreditCards
					color={color}
					height={size}
					testID="credit-cards-icon-test-id"
					width={size}
				/>
			);
		case 'deposits':
			return (
				<Deposit
					color={color}
					height={size}
					testID="deposits-icon-test-id"
					width={size}
				/>
			);
		case 'more':
			return (
				<MoreTiles
					color={color}
					height={size}
					testID="more-icon-test-id"
					width={size}
				/>
			);
		case 'dashboard':
			return (
				<Report
					color={color}
					height={size}
					testID="dashboard-icon-test-id"
					width={size}
				/>
			);
		case 'notifications':
			return (
				<NotificationBell
					color={color}
					height={size}
					testID="notifications-icon-test-id"
					width={size}
				/>
			);
		case 'loans':
			return (
				<LoanWallet
					color={color}
					height={size}
					testID="loans-icon-test-id"
					width={size}
				/>
			);
		case 'paypal':
			return (
				<Paypal
					color={color}
					height={size}
					testID="paypal-icon-test-id"
					width={size}
				/>
			);
		case 'amazon':
			return (
				<Amazon
					color={color}
					height={size}
					testID="amazon-icon-test-id"
					width={size}
				/>
			);
		case 'money-transfer':
			return (
				<MoneyTransfer
					color={color}
					height={size}
					testID="money-transfer-icon-test-id"
					width={size}
				/>
			);
		case 'flag-usa':
			return (
				<FlagUSA
					color={color}
					height={size}
					testID="flag-usa-icon-test-id"
					width={size}
				/>
			);
		case 'search':
			return (
				<Search
					color={color}
					height={size}
					testID="search-icon-test-id"
					width={size}
				/>
			);
		case 'home':
			return (
				<Home
					color={color}
					height={size}
					testID="home-icon-test-id"
					width={size}
				/>
			);
		case 'atm':
			return (
				<Atm
					color={color}
					height={size}
					testID="atm-icon-test-id"
					width={size}
				/>
			);
		case 'ebay':
			return (
				<Ebay
					color={color}
					height={size}
					testID="ebay-icon-test-id"
					width={size}
				/>
			);
		case 'info':
			return (
				<Info
					color={color}
					height={size}
					testID="info-icon-test-id"
					width={size}
				/>
			);
		// no default
	}
};
