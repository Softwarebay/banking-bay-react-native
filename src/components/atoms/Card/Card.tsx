import { StyleProp, ViewStyle } from 'react-native';

import { Flex } from '../Flex/Flex';

import { styles } from './Card.styles';

export interface Props {
	children: React.ReactNode;
	style?: StyleProp<ViewStyle>;
	testID?: string;
}

export const Card = ({ children, style, testID }: Props) => {
	return (
		<Flex direction="row" style={[styles.card, style]} testID={testID}>
			{children}
		</Flex>
	);
};
