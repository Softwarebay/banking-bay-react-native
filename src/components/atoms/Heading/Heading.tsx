import { StyleProp, Text, TextStyle } from 'react-native';

import { styles } from './Heading.styles';

export interface Props {
	text: string;
	size: 'h1' | 'h2' | 'h3' | 'h4' | 'h5' | 'h6';
	color?: string;
	style?: StyleProp<TextStyle>;
	testID?: string;
}

export const Heading = ({ text, size, color, style, testID }: Props) => {
	return (
		<Text
			style={[color ? { color } : styles.default, styles[size], style]}
			testID={testID}
		>
			{text}
		</Text>
	);
};
