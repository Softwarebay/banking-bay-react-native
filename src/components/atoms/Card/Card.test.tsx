import { render, screen } from '@testing-library/react-native';
import testRenderer from 'react-test-renderer';

import { colors } from 'theme';
import { Heading } from '../Heading/Heading';
import { Icon } from '../Icon/Icon';

import type { Props } from './Card';
import { Card } from './Card';

describe.only('Card', () => {
	const props: Props = {
		testID: 'card-test-id',
		style: {
			height: 100,
		},
		children: (
			<>
				<Heading size="h6" text="Text inside Card" />
				<Icon
					color={colors.mainDark}
					name="check"
					size={20}
					testID="card-icon-test-id"
				/>
			</>
		),
	};

	it('should match snapshot', () => {
		const tree = testRenderer.create(<Card {...props} />).toJSON();

		expect(tree).toMatchSnapshot();
	});

	it('should render children in row', () => {
		render(<Card {...props} />);

		const card = screen.getByTestId('card-test-id');
		const cardText = screen.getByText('Text inside Card');
		const cardIcon = screen.getByTestId('card-icon-test-id');

		expect(cardText).toBeTruthy();
		expect(cardIcon).toBeTruthy();

		expect(card).toHaveStyle({ flexDirection: 'row', height: 100 });
	});
});
