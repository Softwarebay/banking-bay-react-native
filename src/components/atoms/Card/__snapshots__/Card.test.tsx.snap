// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card should match snapshot - column 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
      },
      Object {
        "flexDirection": "column",
      },
      Array [
        Object {
          "backgroundColor": "#FFFFFF",
          "borderRadius": 10,
          "shadowColor": "#161966",
          "shadowOffset": Object {
            "height": 6,
            "width": 0,
          },
          "shadowOpacity": 0.08,
          "shadowRadius": 16,
        },
        Object {
          "height": 100,
        },
      ],
    ]
  }
  testID="card-test-id"
>
  <Text
    style={
      Array [
        Object {
          "color": "#1B1D4D",
        },
        Object {
          "fontFamily": "MulishSemiBold",
          "fontSize": 16,
          "lineHeight": 25.6,
        },
        undefined,
      ]
    }
  >
    Text inside Card
  </Text>
  <SvgMock
    color="#1B1D4D"
    height={20}
    testID="check-icon-test-id"
    width={20}
  />
</View>
`;

exports[`Card should match snapshot - row 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
      },
      Object {
        "flexDirection": "row",
      },
      Array [
        Object {
          "backgroundColor": "#FFFFFF",
          "borderRadius": 10,
          "shadowColor": "#161966",
          "shadowOffset": Object {
            "height": 6,
            "width": 0,
          },
          "shadowOpacity": 0.08,
          "shadowRadius": 16,
        },
        Object {
          "height": 100,
        },
      ],
    ]
  }
  testID="card-test-id"
>
  <Text
    style={
      Array [
        Object {
          "color": "#1B1D4D",
        },
        Object {
          "fontFamily": "MulishSemiBold",
          "fontSize": 16,
          "lineHeight": 25.6,
        },
        undefined,
      ]
    }
  >
    Text inside Card
  </Text>
  <SvgMock
    color="#1B1D4D"
    height={20}
    testID="check-icon-test-id"
    width={20}
  />
</View>
`;
